app/src/main/java/org/oppia/android/app/survey/SelectedAnswerAvailabilityReceiver.kt

```diff
@@ -0,0 +1,22 @@
+package org.oppia.android.app.survey
+
+import org.oppia.android.app.model.SurveySelectedAnswer
+
+/**
+ *  A handler for receiving any change in answer availability to update the 'next' button.
+ */
+interface SelectedAnswerAvailabilityReceiver {
+  /**
+   *  Called when the input answer availability changes.
+   */
+  fun onPendingAnswerAvailabilityCheck(inputAnswerAvailable: Boolean)
+}
+
+/** A callback that will be called when a user submits an answer. */
+interface SelectedAnswerHandler {
+  /** Return the current selected answer that is ready for submission. */
+  fun getPendingAnswer(selectedAnswer: SurveySelectedAnswer)
```

Comment:
I renamed pending to getMultipleChoiceAnswer() because that is the distinction that I had hoped to make. One of the listeners is called when a multiplechoice question needs to be submitted, and the other for free form answers.